internal/controller/storageclient_controller.go

```diff
@@ -239,6 +241,23 @@ func (r *storageClientReconcile) reconcile(ctx context.Context, req ctrl.Request
 
 	result, reconcileErr := r.reconcilePhases()
 
+	storageClaimList := &metav1.PartialObjectMetadataList{}
+	storageClaimList.SetGroupVersionKind(v1alpha1.GroupVersion.WithKind("StorageClaimList"))
+	if err := r.list(storageClaimList); err != nil {
+		r.log.Info("No StorageClaims remainings")
+	}
+
+	for _, storageClaim := range storageClaimList.Items {
+		if err := r.Client.Delete(r.ctx, &storageClaim); client.IgnoreNotFound(err) != nil {
+			r.log.Error(err, "Failed to delete StorageClaim", "name", storageClaim.Name)
+		}
+
+		controllerutil.RemoveFinalizer(&storageClaim, StorageClaimFinalizer)
```

Comment:
can you check if this returns a bool? if yes, update only if we actually removed the finalizer on the im-mem obj.